evaluate/task/task-transpile.go

```diff
@@ -0,0 +1,178 @@
+type TaskArgumentsTranspile struct {
+	// TargetLanguage holds the target language for transpilation.
+	TargetLanguage language.Language
+	// StubFilePath holds the path for the file containing just the function signature of the language we are transpiling to.
+	StubFilePath string
+}
+
+var _ evaltask.Task = (*TaskTranspile)(nil)
+
+// Identifier returns the transpilation task identifier.
+func (t *TaskTranspile) Identifier() evaltask.Identifier {
+	return IdentifierTranspile
+}
+
+// Run transpiles code between Go and Java and runs predefined tests to check if the transpilation was successful.
+func (t *TaskTranspile) Run(ctx evaltask.Context) (repositoryAssessment map[evaltask.Identifier]metrics.Assessments, problems []error, err error) {
+	modelCapability, ok := ctx.Model.(model.CapabilityTranspile)
+	if !ok {
+		return nil, nil, pkgerrors.Wrap(evaltask.ErrTaskUnsupportedByModel, fmt.Sprintf("%q does not support %q", ctx.Model.ID(), string(t.Identifier())))
+	}
+
+	taskLogger, err := newTaskLogger(ctx, t)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer func() {
+		taskLogger.finalize(problems)
+	}()
+
+	var packagePaths []string
+	files, err := os.ReadDir(ctx.Repository.DataPath())
+	if err != nil {
+		return nil, nil, pkgerrors.WithStack(err)
+	}
+	for _, file := range files {
+		if file.IsDir() && !strings.HasPrefix(file.Name(), ".") { // Ignore hidden directories.
+			packagePaths = append(packagePaths, filepath.Join(ctx.Repository.DataPath(), file.Name()))
+		}
+	}
+
+	modelAssessments := metrics.NewAssessments()
+	for _, packagePath := range packagePaths {
+		if err := ctx.Repository.Reset(ctx.Logger); err != nil {
+			ctx.Logger.Panicf("ERROR: unable to reset temporary repository path: %s", err)
+		}
+
+		var targetLanguage language.Language
+		if _, ok := ctx.Language.(*golang.Language); ok {
+			targetLanguage = &java.Language{}
+		} else {
+			targetLanguage = &golang.Language{}
+		}
+
+		sourceFilePath, stubFilePath, err := t.unpackTranspilerPackage(ctx, taskLogger.Logger, targetLanguage, packagePath)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		modelContext := model.Context{
+			Language: ctx.Language,
+
+			RepositoryPath: packagePath,
+			FilePath:       sourceFilePath,
+
+			Arguments: &TaskArgumentsTranspile{
+				TargetLanguage: targetLanguage,
+				StubFilePath:   stubFilePath,
+			},
+
+			Logger: taskLogger.Logger,
+		}
+		assessments, err := modelCapability.Transpile(modelContext)
+		if err != nil {
+			problems = append(problems, pkgerrors.WithMessage(err, sourceFilePath))
+
+			continue
+		}
+		if assessments[metrics.AssessmentKeyProcessingTime] == 0 {
+			return nil, nil, pkgerrors.Errorf("no model response time measurement present for %q at repository %q", ctx.Model.ID(), ctx.Repository.Name())
+		}
+		modelAssessments.Add(assessments)
+		modelAssessments.Award(metrics.AssessmentKeyResponseNoError)
+
+		coverage, ps, err := targetLanguage.Execute(taskLogger.Logger, packagePath)
```

Comment:
Mhm yeah now that I see this I am starting to wonder. Assuming this is `golang/transpile/...` then we are actually here doing Java execution. Which means we are scoring the transpilation from Go to Java. But in the results it will end up alongside the Go scores. I think maaaybe we switch the whole thing around... an it should be about transpiling from some other language to **this** language and not from **this** language to another one. So basically you would switch around the `testdata` and also make this `ctx.Language.Execute`... sorry :sob: